### PR TITLE
SFK: Improve Arugal intro event

### PR DESCRIPTION
**Pull request by schell244**
## 🍰 Pullrequest
- ~~fix Arugal not being invisible before script starts~~
- adapt timers for his script actions
- use correct spell for visual teleporting https://classic.wowhead.com/npc=10000/arugal#abilities
- despawn Arugal in script instead of making him invisible in cpp 

Note: making Vincent "die" during the script is different from the video, but was part of the script before - so I kept it in.

### Proof
- https://youtu.be/6P2ZrJzuBdY?t=633

### Issues
- None

### How2Test
- enter Shadowfang Keep and watch intro event  at the entrance
- to ensure that the event only happens once and Arugal is despawned: re-enter or relog in dungeon

### Todo / Checklist
- [X] None

### CODE REVIEW

**Comment by ratkosrb** [comment]
> fix Arugal not being invisible before script starts

But he is already invisible when you enter.
https://streamja.com/BO01n

**Comment by schell244** [comment]
Huh, than this was probably the case on CMangos. I've checked the event on both CMangos and VMangos and might have mixed  up where he was not visible initially.

**Comment by ratkosrb** [comment]
> making Vincent "die" during the script is different from the video

In sniffs he has stand_state = 0  (UNIT_STAND_STATE_STAND), so his stand state changes to dead later by a script.

**Comment by schell244** [comment]
On current VMangos he says his first text before he is visible (and has himself teleported in). I will update the description.

File: sql/migrations/20220721212432_world.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+SET v = (SELECT COUNT(*) FROM `migrations` WHERE `id`='20220721212432');
+IF v=0 THEN
+INSERT INTO `migrations` VALUES ('20220721212432');
+-- Add your query below.
+
+-- SFK: Arugal intro script
+DELETE FROM `generic_scripts` WHERE `id`=10000;
+INSERT INTO `generic_scripts` (`id`, `delay`, `priority`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(10000, 0, 0, 4, 46, 134217728, 1, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'SFK Intro - Arugal Invisible'),
+(10000, 5, 0, 4, 46, 134217728, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'SFK Intro - Arugal Visible'),
```
Comment (ratkosrb) [inline sql/migrations/20220721212432_world.sql:15]: Why are you toggling UNIT_FLAG_AURAS_VISIBLE ???
That does not make units invisible. It's for showing auras to hostile players. This flag is applied when you are afflicted by a mage's Detect Magic debuff.

**Comment by ratkosrb** [comment]
> On current VMangos he says his first text before he is visible (and has himself teleported in). I will update the description.

That's intended. He yells first, and then teleports.

**Comment by schell244** [comment]
> > On current VMangos he says his first text before he is visible (and has himself teleported in). I will update the description.
> 
> That's intended. He yells first, and then teleports.

Ok, in this case my PR is pointless. Thanks for checking.

**Pull request closed without merging**